enumerables.rb

```diff
@@ -0,0 +1,184 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/BlockNesting
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < length
+        if instance_of?(Array)
+          yield self[i]
+        else
+          yield(keys[i], values[i])
+        end
+        i += 1
+      end
+    else
+      puts 'block was not given'
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < length
+        if instance_of?(Array)
+          yield(self[i], i)
+        else
+          yield(values[i], i)
+        end
+        i += 1
+      end
+    else
+      puts 'block was not given'
+    end
+  end
+
```

Comment:
- [ ] enumerables #my_each_with_index returns an enumerator if no block is given
- [ ]  enumerables #my_each_with_index when a block is given when self is an array returns the array itself after calling the given block once for each element in self
- [ ] enumerables #my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self
``` NameError: undefined local variable or method `length' for 5..50:Range ```
- [ ] enumerables #my_each_with_index when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self
``` NameError: undefined local variable or method `length' for 5..50:Range ```
- [ ]  enumerables #my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self
```
expected: "Num: [:a, 1], idx: 0\nNum: [:b, 2], idx: 1\nNum: [:c, 3], idx: 2\nNum: [:d, 4], idx: 3\nNum: [:e, 5], idx: 4\n"
got: "Num: 1, idx: 0\nNum: 2, idx: 1\nNum: 3, idx: 2\nNum: 4, idx: 3\nNum: 5, idx: 4\n"

```